activesupport/lib/active_support/core_ext/range/compare_range.rb

```diff
@@ -17,6 +17,7 @@ def ===(value)
       if value.is_a?(::Range)
         # 1...10 includes 1..9 but it does not include 1..10.
         # 1..10 includes 1...11 but it does not include 1...12.
+        return false if value.first && value.last && value.first > value.last
```

Comment:
Adding this line here separates the comments from the code they're describing, could you move it above them?